### PR TITLE
POR-1633: support logs on new view

### PR DESCRIPTION
**Pull request by d-g-town**
This PR adds support for live and historical logs based on the new App object (rather than helm values).

Logs will not work until a new version of the agent is pushed out.

### CODE REVIEW

File: dashboard/src/main/home/app-dashboard/validate-apply/logs/Logs.tsx [hunk 1]
```diff
@@ -0,0 +1,521 @@
...
+            const agentImage = res.data?.image;
+            if (!isAgentVersionUpdated(agentImage)) {
+                throw new Error("Agent version is not updated");
+            }
+        } catch (err) {
+            if (err.response?.status === 404) {
+                setHasPorterAgent(false);
+            }
+        }
+    };
```
Comment (ianedwards) [inline dashboard/src/main/home/app-dashboard/validate-apply/logs/Logs.tsx:301]: one thing I'm noticing here is that if the request fails for any reason other than 404, `hasPorterAgent` will still be true.

Also tripped me up that this fn is defined after the useEffect where it's called. This is a solid candidate to use wrap in a `useQuery`

File: dashboard/src/main/home/app-dashboard/validate-apply/logs/Logs.tsx [hunk 2]
```diff
@@ -0,0 +1,521 @@
...
+
+        setIsPorterAgentInstalling(true);
+
+        api
+            .installPorterAgent("<token>", {}, { project_id, cluster_id })
+            .then()
+            .catch((err) => {
+                setIsPorterAgentInstalling(false);
+                console.log(err);
+            });
```
Comment (ianedwards) [inline dashboard/src/main/home/app-dashboard/validate-apply/logs/Logs.tsx:315]: `isPorterAgentInstalling` not going back to false unless the request fails?

File: dashboard/src/main/home/app-dashboard/validate-apply/logs/types.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    app_name: z.string(),
+    raw_labels: RawLabelsSchema.optional(),
+});
+
+export const AgentLogSchema = z.object({
+    line: z.string(),
+    timestamp: z.string(),
+    metadata: AgentLogMetadataSchema.optional(),
+});
+export type AgentLog = z.infer<typeof AgentLogSchema>;
```
Comment (ianedwards) [inline dashboard/src/main/home/app-dashboard/validate-apply/logs/types.ts:45]: small nit but I usually `camelCase` the validator and have it end with `Validator` (i.e. agentLogValidator)

Also usually prefer type to interface

**Review by ianedwards — APPROVED** [review]
don't have a ton of context on the logging functionality, just pointing out a few things to double check that I came across!

File: api/server/handlers/porter_app/stream_logs.go [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+	if request.ServiceName != "all" {
+		labels = append(labels, fmt.Sprintf("%s=%s", lokiLabel_PorterServiceName, request.ServiceName))
+	}
+
+	err = agent.StreamPorterAgentLokiLog(labels, string(startTime), request.SearchParam, 0, safeRW)
+
+	if err != nil {
+		c.HandleAPIError(w, r, apierrors.NewErrInternal(err))
+		return
+	}
```
Comment (ferozemohideen) [inline api/server/handlers/porter_app/stream_logs.go:137]: gotta add the telemetry if you're gonna copy and paste :')

File: dashboard/src/main/home/app-dashboard/validate-apply/logs/utils.ts [hunk 5]
```diff
@@ -0,0 +1,477 @@
...
+      return {
+        line: Anser.ansiToJson(log.toString()),
+        lineNumber: idx + 1,
+        timestamp: undefined,
+      }
+    }
+  });
+};
+
+export const useLogs = (
```
Comment (ferozemohideen) [inline dashboard/src/main/home/app-dashboard/validate-apply/logs/utils.ts:39]: are we copying this hook for the third time lol

**Review by ferozemohideen — CHANGES_REQUESTED** [review]

**Review by ferozemohideen — APPROVED** [review]

**Pull request merged**